example/sysinfo/odp_sysinfo.c

```diff
@@ -361,7 +380,209 @@ static void print_auth(odp_auth_alg_t alg)
+		printf("    lso.proto.sctp_ipv6:           %u\n", capa->lso.proto.sctp_ipv6);
+		printf("    maxlen.equal:                  %i\n", capa->maxlen.equal);
+		printf("    maxlen.min_input:              %u B\n", capa->maxlen.min_input);
+		printf("    maxlen.max_input:              %u B\n", capa->maxlen.max_input);
+		printf("    maxlen.min_output:             %u B\n", capa->maxlen.min_output);
+		printf("    maxlen.max_output:             %u B\n", capa->maxlen.max_output);
+		printf("    max_tx_aging_tmo_ns:           %" PRIu64 " ns\n",
+		       capa->max_tx_aging_tmo_ns);
+		printf("    tx_compl.queue_type_sched:     %i\n", capa->tx_compl.queue_type_sched);
+		printf("    tx_compl.queue_type_plain:     %i\n", capa->tx_compl.queue_type_plain);
+		printf("    tx_compl.mode_event:           %u\n", capa->tx_compl.mode_event);
+		printf("    tx_compl.mode_poll:            %u\n", capa->tx_compl.mode_poll);
+		printf("    tx_compl.max_compl_id:         %u\n", capa->tx_compl.max_compl_id);
+		printf("    free_ctrl.dont_free:           %u\n", capa->free_ctrl.dont_free);
+		printf("    reassembly.ip:                 %i\n", capa->reassembly.ip);
+		printf("    reassembly.ipv4:               %i\n", capa->reassembly.ipv4);
+		printf("    reassembly.ipv6:               %i\n", capa->reassembly.ipv6);
+		printf("    reassembly.max_wait_time:      %" PRIu64 " ns\n",
+		       capa->reassembly.max_wait_time);
+		printf("    reassembly.max_num_frags:      %u\n", capa->reassembly.max_num_frags);
+		printf("    stats.pktio:                   0x%" PRIx64 "\n",
+		       capa->stats.pktio.all_counters);
+		printf("    stats.pktin_queue:             0x%" PRIx64 "\n",
+		       capa->stats.pktin_queue.all_counters);
+		printf("    stats.pktout_queue:            0x%" PRIx64 "\n",
+		       capa->stats.pktout_queue.all_counters);
+		printf("    flow_control.pause_rx:         %u\n", capa->flow_control.pause_rx);
+		printf("    flow_control.pfc_rx:           %u\n", capa->flow_control.pfc_rx);
+		printf("    flow_control.pause_tx:         %u\n", capa->flow_control.pause_tx);
+		printf("    flow_control.pfc_tx:           %u\n", capa->flow_control.pfc_tx);
+
+		free(appl_args->pktio[i].name);
+	}
+}
+
+static void usage(void)
+{
+	printf("\n"
+	       "System Information\n"
+	       "\n"
+	       "Usage: %s OPTIONS\n"
+	       "  E.g. %s -i eth0\n"
+	       "\n"
+	       "Optional OPTIONS:\n"
+	       "  -i, --interfaces   Ethernet interfaces for packet I/O, comma-separated, no\n"
+	       "                     spaces.\n"
+	       "  -h, --help         Display help and exit.\n"
+	       "\n", PROG_NAME, PROG_NAME);
+}
+
+static void parse_interfaces(appl_args_t *config, const char *optarg)
+{
+	char *tmp_str = strdup(optarg), *tmp;
+
+	if (tmp_str == NULL)
+		return;
+
+	tmp = strtok(tmp_str, ",");
+
+	while (tmp && config->num_pktio < MAX_IFACES) {
+		config->pktio[config->num_pktio].name = strdup(tmp);
```

Comment:
To avoid memory leak possibility (strdup() here and free() elsewhere), memory for pktio name could have a static size (e.g. 128 chars) and just strncpy() here.